### PR TITLE
DOC: Fix pandas.index.copy summary documentation

### PR DESCRIPTION
**Pull request by asepwhite**
- [ ] closes https://github.com/pandanistas/pandanistas_sprint_jakarta2020/issues/8
- [ ] tests added / passed
- [ ] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry
Output of `python scripts/validate_docstrings.py pandas.Index.copy`

################################################################################
################################################################################

Make a copy of this object.

Name and dtype sets those attributes on the new object.

Parameters
----------
name : Label, optional
    Set name for new object.
deep : bool, default False
dtype : numpy dtype or pandas type, optional
    Set dtype for new object.
names : list-like, optional
    Kept for compatibility with MultiIndex. Should not be used.

Returns
-------
Index
    Index refer to new object which is a copy of this object.

Notes
-----
In most cases, there should be no functional difference from using
``deep``, but if ``deep`` is passed it will attempt to deepcopy.

################################################################################
################################################################################

3 Errors found:
        Parameter "deep" has no description
        See Also section not found
        No examples section found

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @asepwhite !

In a follow up you can replace `Label` by `label` as the type, and add the description to `deep` if you want.

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -825,20 +825,24 @@ def repeat(self, repeats, axis=None):
...
         deep : bool, default False
         dtype : numpy dtype or pandas type, optional
+            Set dtype for new object.
         names : list-like, optional
             Kept for compatibility with MultiIndex. Should not be used.
 
         Returns
         -------
         Index
+            Index refer to new object which is a copy of this object.
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py]: less verbose here: "New Index object"

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -825,20 +825,24 @@ def repeat(self, repeats, axis=None):
...
+        Name and dtype sets those attributes on the new object.
 
         Parameters
         ----------
-        name : Label
+        name : Label, optional
+            Set name for new object.
         deep : bool, default False
         dtype : numpy dtype or pandas type, optional
+            Set dtype for new object.
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py]: "Set" her and on 835 are unnecessary

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -825,20 +825,24 @@ def repeat(self, repeats, axis=None):
...
+
+        Name and dtype sets those attributes on the new object.
 
         Parameters
         ----------
-        name : Label
+        name : Label, optional
+            Set name for new object.
         deep : bool, default False
         dtype : numpy dtype or pandas type, optional
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py]: "pandas type" isnt a thing, possibly "ExtensionDType"?  i think strs are also accepted if they can be mapped to dtypes

**Comment by datapythonista** [comment]
Thanks @asepwhite. If you can please address the review comments in a follow up PR that would be great.

**Pull request merged**